dvc/ignore.py

```diff
@@ -71,10 +71,25 @@ def __eq__(self, other):
         return self.basenames == other.basenames
 
 
+class DvcIgnoreRepo(DvcIgnore):
```

Comment:
Would be interesting to see perf comparison with and without this change when you have giant directories in your repo.

Reply:
@pared ^^

Reply:
Little summary of performed tests:
created directory with 10 directories, each having 10 directories and so on.
The tree had 6 such levels so alltogether at the bottom there were 1M dirs.
Adding new Ignore filter increased stages gathering time from `7m 58s` to `8m 15s` which is roughly 3%.  It is not incosiderable, but I would say that in case of sucha big project user will probably want to leverage the `.dvcignore` functionality to reduce this time by omitting different dirs.